core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java

```diff
@@ -1555,5 +1561,42 @@ private void setInterruptFlagInternal(AtomicBoolean flag) {
     public void setCacheProvider(CacheProvider cacheProvider) {
       reader.setCacheProvider(cacheProvider);
     }
+
+    /**
+     * Returns an estimate of the number of entries that overlap the given extent. This is an
+     * estimate because the extent may or may not entirely overlap with each of the index entries
+     * included in the count. Will never underestimate but may overestimate.
+     *
+     * @param extent the key extent
+     * @return the estimate
+     */
+    @Override
+    public long estimateEntries(KeyExtent extent) throws IOException {
+      long totalEntries = 0;
+      Key startKey = extent.toDataRange().getStartKey();
+      IndexEntry indexEntry;
+
+      for (LocalityGroupReader lgr : currentReaders) {
+        boolean prevEntryOverlapped = false;
+        var indexIter = startKey == null ? lgr.getIndex() : lgr.index.lookup(startKey);
+
+        while (indexIter.hasNext()) {
+          indexEntry = indexIter.next();
+          if (extent.contains(indexEntry.getKey().getRow())) {
+            totalEntries += indexEntry.getNumEntries();
+            prevEntryOverlapped = true;
+          } else if (prevEntryOverlapped) {
+            // The last index entry included in the count is the one after the last contained by the
+            // extent. This is because it is possible for the extent to overlap this index entry
+            // but there is no way to check whether it does or not. The index entry only contains
+            // info about the last key, but the extent may overlap but not with the last key.
+            totalEntries += indexEntry.getNumEntries();
+            prevEntryOverlapped = false;
```

Comment:
could break out early here. 

```suggestion
            prevEntryOverlapped = false;
            break;
```